src/fides/api/service/privacy_request/dsr_package/dsr_report_builder.py

```diff
@@ -195,27 +190,59 @@ def _write_attachment_content(
                 continue
 
             file_name = attachment.get("file_name")
-            if not file_name:
-                logger.warning("Skipping attachment with no file name")
-                continue
-
             download_url = attachment.get("download_url")
-            if not download_url:
-                logger.warning("Skipping attachment with no download URL")
+            if not file_name or not download_url:
+                logger.warning(
+                    f"Skipping attachment with missing {'file name' if not file_name else 'download URL'}"
+                )
                 continue
 
+            # Format file size
             file_size = attachment.get("file_size")
-            if isinstance(file_size, (int, float)):
-                file_size = format_size(float(file_size))
+            file_size = (
+                format_size(float(file_size))
+                if isinstance(file_size, (int, float))
+                else "Unknown"
+            )
+
+            # Get or generate unique filename
+            attachment_key = (download_url, file_name)
+            if attachment_key not in self.processed_attachments:
+                used_filenames = (
+                    self.used_filenames_attachments
+                    if dataset_name == "attachments"
+                    else self.used_filenames_data
+                )
+                unique_filename = get_unique_filename(file_name, used_filenames)
+                used_filenames.add(unique_filename)
+                self.processed_attachments[attachment_key] = unique_filename
+            else:
+                unique_filename = self.processed_attachments[attachment_key]
+
+            # Set attachment URL based on streaming mode
+            if self.enable_streaming:
+                depth = directory.count("/") + 1 if directory.startswith("data/") else 0
+                attachment_url = (
+                    "../" * depth + f"attachments/{unique_filename}"
+                    if depth
+                    else unique_filename
+                )
```

Comment:
Thought for the future, we may be able to obviate the need for the calculation by setting a `<base href="" />` in the document and then references `attachments/{unique_filename}` directly, but for now this totally works. 👍🏻